crates/bevy_pbr/src/render/pbr.wgsl

```diff
@@ -367,9 +374,47 @@ fn directional_light(light: DirectionalLight, roughness: f32, NdotV: f32, normal
 fn fetch_point_shadow(light_id: u32, frag_position: vec4<f32>, surface_normal: vec3<f32>) -> f32 {
     let light = point_lights.data[light_id];
 
+    let surface_to_light = light.position_radius.xyz - frag_position.xyz;
+
+    if ((light.flags & POINT_LIGHT_FLAGS_IS_SPOTLIGHT_BIT) != 0u) {
+        // construct the light view matrix
+        // view matrix z_axis is the reverse of transform.forward()
+        let fwd = -light.light_custom_data.xyz;
+        let distance_to_light = dot(fwd, surface_to_light);
+        let offset_position = -surface_to_light + (light.shadow_depth_bias * normalize(surface_to_light)) + (surface_normal.xyz * light.shadow_normal_bias) * distance_to_light;
+
+        // the construction of the up and right vectors needs to precisely mirror the code 
+        // in render/light.rs:spotlight_view_matrix
+        var sign = -1.0;
+        if (fwd.z >= 0.0) {
+            sign = 1.0;
+        }
+        let a = -1.0 / (fwd.z + sign);
+        let b = fwd.x * fwd.y * a;
+        let up_dir = vec3<f32>(1.0 + sign * fwd.x * fwd.x * a, sign * b, -sign * fwd.x);
+        let right_dir = vec3<f32>(-b, -sign - fwd.y * fwd.y * a, fwd.y);
+        let light_inv_rot = mat3x3<f32>(right_dir, up_dir, fwd);
+        let projected_position = offset_position * light_inv_rot;
```

Comment:
basically because the matrix is a pure rotation matrix, the inverse is just the transpose, and to calculate the product of the transpose with a vector we can just pre-multiply instead of post-multplying. i like this because then the code for the matrix construction can be identical between CPU and GPU.

if you're happy with that i will add a comment to that effect.

Reply:
Or post-multiply instead of pre-multiply in this case? Sounds good.